hw/xwin/dri/windowsdri.c

```diff
@@ -46,102 +46,77 @@ WindowsDRIResetProc(ExtensionEntry* extEntry)
 static int
 ProcWindowsDRIQueryVersion(ClientPtr client)
 {
-    xWindowsDRIQueryVersionReply rep;
-
     REQUEST_SIZE_MATCH(xWindowsDRIQueryVersionReq);
-    rep.type = X_Reply;
-    rep.length = 0;
-    rep.sequenceNumber = client->sequence;
-    rep.majorVersion = SERVER_WINDOWSDRI_MAJOR_VERSION;
-    rep.minorVersion = SERVER_WINDOWSDRI_MINOR_VERSION;
-    rep.patchVersion = SERVER_WINDOWSDRI_PATCH_VERSION;
+
+    xWindowsDRIQueryVersionReply reply = {
+        .majorVersion = SERVER_WINDOWSDRI_MAJOR_VERSION,
+        .minorVersion = SERVER_WINDOWSDRI_MINOR_VERSION,
+        .patchVersion = SERVER_WINDOWSDRI_PATCH_VERSION,
+    };
+
     if (client->swapped) {
-        swaps(&rep.sequenceNumber);
-        swapl(&rep.length);
-        swaps(&rep.majorVersion);
-        swaps(&rep.minorVersion);
-        swapl(&rep.patchVersion);
+        swaps(&reply.majorVersion);
+        swaps(&reply.minorVersion);
+        swapl(&reply.patchVersion);
     }
-    WriteToClient(client, sizeof(xWindowsDRIQueryVersionReply), &rep);
-    return Success;
+
+    return X_SEND_REPLY_SIMPLE(client, reply);
 }
 
 static int
 ProcWindowsDRIQueryDirectRenderingCapable(ClientPtr client)
 {
-    xWindowsDRIQueryDirectRenderingCapableReply rep;
 
     REQUEST(xWindowsDRIQueryDirectRenderingCapableReq);
     REQUEST_SIZE_MATCH(xWindowsDRIQueryDirectRenderingCapableReq);
-    rep.type = X_Reply;
-    rep.length = 0;
-    rep.sequenceNumber = client->sequence;
+
+    xWindowsDRIQueryDirectRenderingCapableReply reply = { 0 };
 
     if (!client->local)
-        rep.isCapable = 0;
+        reply.isCapable = 0;
     else
-        rep.isCapable = glxWinGetScreenAiglxIsActive(screenInfo.screens[stuff->screen]);
+        reply.isCapable = glxWinGetScreenAiglxIsActive(screenInfo.screens[stuff->screen]);
```

Comment:
The branch where `isCapable` is set to zero is not really needed, so this `if` can be a bit simplified.